cargo-near-build/src/fs.rs

```diff
@@ -4,15 +4,45 @@ use eyre::WrapErr;
 /// Copy a file to a destination.
 ///
 /// Does nothing if the destination is the same as the source to avoid truncating the file.
-pub fn copy(from: &Utf8Path, to: &Utf8Path) -> eyre::Result<Utf8PathBuf> {
-    let out_path = to.join(from.file_name().unwrap());
+pub fn copy(from: &Utf8Path, out_dir: &Utf8Path) -> eyre::Result<Utf8PathBuf> {
+    if !out_dir.is_dir() {
+        return Err(eyre::eyre!("`{}` should be a directory", out_dir));
+    }
+    let out_path = out_dir.join(from.file_name().unwrap());
+    tracing::debug!("Copying file `{}` -> `{}`", from, out_path,);
     if from != out_path {
         std::fs::copy(from, &out_path)
             .wrap_err_with(|| format!("failed to copy `{}` to `{}`", from, out_path))?;
     }
     Ok(out_path)
 }
 
+/// Copy a file to a file destination.
+///
+/// Does nothing if the destination is the same as the source to avoid truncating the file.
+pub fn copy_to_file(from: &Utf8Path, to: &Utf8Path) -> eyre::Result<Utf8PathBuf> {
+    tracing::debug!("Copying file `{}` -> `{}`", from, to,);
+
+    if from != to && to.is_file() {
```

Comment:
Does the method work with directories as well? Maybe use `if !from.is_file() || !to.is_file() -> return`